vcs/github/github.go

```diff
@@ -299,16 +304,90 @@ func (w *workingCopy) checkout(ctx context.Context, version vcs.VersionNumber) e
 
 		return err
 	}
-	w.version = version
-	return nil
+	for {
+		err := w.g.git(ctx, w.dir, nil, "checkout", string(version))
+		if err == nil {
+			w.version = version
+			return nil
+		}
+		var exitErr *exec.ExitError
+		if !errors.As(err, &exitErr) {
+			return handleVersionNotFound(err)
+		}
+		if exitErr.ExitCode() != 128 {
+			return handleVersionNotFound(err)
+		}
+		tries += 1
+		if tries > maxTries {
+			return fmt.Errorf("git checkout exited with a 128 error code %d times in a row, aborting (%w)", maxTries, err)
+		}
+		select {
+		case <-ctx.Done():
+			return err
+		case <-time.After(100 * time.Millisecond):
```

Comment:
nit: time.After could negatively affect memory consumption in go versions before 1.23

https://pkg.go.dev/time#After